x/evm/handler.go

```diff
@@ -73,14 +77,17 @@ func handleETHTxMsg(ctx sdk.Context, k Keeper, msg types.EthereumTxMsg) sdk.Resu
 	k.CommitStateDB.Prepare(ethHash, common.Hash{}, k.TxCount.Get())
 	k.TxCount.Increment()
 
-	bloom, res := st.TransitionCSDB(ctx)
-	if res.IsOK() {
-		k.Bloom.Or(k.Bloom, bloom)
+	returnData := st.TransitionCSDB(ctx)
+	if returnData.Result.IsOK() {
+		k.Bloom.Or(k.Bloom, returnData.Bloom)
+		fmt.Println(returnData.Logs[0])
+		k.Logs = append(k.Logs, returnData.Logs...) // does k.SetBlockLogs or k.AddLog need to be called here?
```

Comment:
yes, it should. Setting logs on the keeper will do nothing. The reason that a bloom filter is attached to the keeper is because it is the cumulative bloom filter for the block that needs to be updated after each state transition, and is persisted at the end of the block (or more specifically at the start of the next block)

Reply:
that makes sense, so the bloom in the keeper is actually the block bloom, not the tx bloom? 

is there a way to get the tx hash for the transition at this point? it would make more sense to store tx hash -> logs than block -> all block logs

Reply:
Correct.

For the second point, yes it is possible. When doing this previously I thought it wasn't available and was rehashing the Ethereumtx but looking back I see the tx bytes are attached to the sdk `Context`. (I don't know how I missed that) https://github.com/ChainSafe/ethermint/blob/e66bf4c38b05e59320f5e9d6f0c309cf171eec58/x/evm/handler.go#L56 Existing logic is done there and can be used as reference, but can be cleaned up since I'm rehashing the tx unnecessarily.

Reply:
cool thanks, I'll update this to save the logs for each tx in the KVStore

Reply:
> yes, it should. Setting logs on the keeper will do nothing.

++ Keeper ≠ store